syntax = "proto3";

package wood.mike.greet;

service Greeter {
  //Unary
  rpc SayHello (HelloRequest) returns (HelloReply);

  // server streaming
  rpc ParrotSaysHello (HelloRequest) returns (stream HelloReply);

  // client streaming
  rpc ChattyClientSaysHello (stream HelloRequest) returns (DelayedReply);

  // both streaming
  rpc InteractingHello (stream HelloRequest) returns (stream HelloReply);
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
  string greeting = 2;
}

// The response message containing the greetings.
message HelloReply {
  string message = 1;
}

message DelayedReply {
	string message = 1;
	repeated HelloRequest request = 2;
}